cmd/maya-apiserver/app/server/snapshot_endpoint_v1alpha1.go

```diff
@@ -0,0 +1,171 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/types/v1"
+	"github.com/openebs/maya/volume/provisioners/jiva"
+)
+
+// SnapshotSpecificRequest deals with snapshot API request w.r.t a Volume
+func (s *HTTPServer) snapshotV1alpha1SpecificRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	// Extract info from path after trimming
+	path := strings.TrimPrefix(req.URL.Path, "/latest/snapshots")
+
+	// Check request validity
+	if path == req.URL.Path {
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+	volOp := &volumeAPIOpsV1alpha1{
+		req:  req,
+		resp: resp,
+	}
+	switch {
+	case strings.Contains(path, "/create/"):
+		return volOp.snapshotCreate()
+	case strings.Contains(path, "/revert/"):
+		return volOp.snapshotRevert()
+	case strings.Contains(path, "/list"):
+		volName := strings.TrimPrefix(path, "/list/")
+		return volOp.snapshotList(volName)
+	default:
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+}
+
+// SnapshotCreate is http handler which handles snaphsot-create request
+func (v *volumeAPIOpsV1alpha1) snapshotCreate() (interface{}, error) {
```

Comment:
Do we need to make use of volumeAPIOpsV1alpha1 struct?
Can we make use of a snapshot specific struct ?
This struct should be specific to jiva or cstor & in turn should implement a generic function/interface.

Reply:
How does the logic learn if the snapshot will be a jiva based snapshot vs cstor based snapshot?